### PR TITLE
Feat: Dark mode

### PR DESCRIPTION
**Pull request by sananand007**
## Description
Enabling Dark Mode for unwind.dev

There is no Dark Mode feature on current Production. We add a theme switcher button on the Home Page that enables the Dark Mode for Users 

We will handle the below endpoints with this feature only as these are our 99% used endpoints
- Main Landing Page 
- Plan template Page or search results page
- About 

## Solution

- [x] Use SVG to theme switch
- [x] use data attributes to simplify CSS for light and dark mode
- [x] complete switch using the theme switcher svg
- [x] make sure we persist using local storage
- [x] migrate the theme to other crucial pages / persist
- [x] Test and deploy

## Testing
- [x] Integration testing on Heroku staging
- [ ] Added new unit tests

## Checks
- [ ] Have you removed commented code?
- [ ] Have you used gofmt to format your code?

### CODE REVIEW

File: assets/js/mode-switch.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+    // Set the new HTML attribute
+    rootElem.setAttribute('data-theme', newTheme)
+
+    // Set the new Local Storage item
+    localStorage.setItem('theme', newTheme)
+}
+
+// Add event Listener for the theme switcher
+document.querySelector('#theme-switcher').addEventListener('click', switchTheme)
```
Comment (timwangmusic) [inline assets/js/mode-switch.js:17]: Add white space please

File: assets/templates/login_page.html [hunk 2]
```diff
@@ -169,5 +168,13 @@ <h1>Vacation Planner</h1>
     </div>
 
     <script src="assets/js/login.js" type="module"></script>
+    <script>
```
Comment (timwangmusic) [inline assets/templates/login_page.html:171]: Please refactor and put these in a separate file

File: assets/templates/user_profile.html [hunk 3]
```diff
@@ -35,7 +27,7 @@
     </nav>
     <h1>Profile</h1>
     <br />
-    <div class="container">
+    <div class="container" id="user-profile-container">
```
Comment (timwangmusic) [inline assets/templates/user_profile.html:30]: So we use this ID somewhere?

File: assets/templates/user_profile.html [hunk 4]
```diff
@@ -71,5 +63,10 @@ <h5 class="card-title">Most Searched Place</h5>
     <script src="https://cdn.jsdelivr.net/npm/js-cookie@rc/dist/js.cookie.min.js"></script>
     <script type="module" src="assets/js/user.js"></script>
     <script type="module" src="assets/js/user_profile.js"></script>
+    <script
```
Comment (timwangmusic) [inline assets/templates/user_profile.html:66]: I don't think this change is needed.

File: test/redis_client_mocks/nearby_search_test.go [hunk 5]
```diff
@@ -82,7 +82,7 @@ func TestGetPlaces_shouldExcludePlacesOutsideOfSearchRadius(t *testing.T) {
 
 	cachedEateryPlaces, err := RedisClient.NearbySearch(RedisContext, &placeSearchRequest)
 	if err != nil {
-		t.Errorf(err.Error())
+		t.Errorf("RedisClient error #{err.Error()}")
```
Comment (timwangmusic) [inline test/redis_client_mocks/nearby_search_test.go:85]: Use %s as format directive

**Review by timwangmusic — CHANGES_REQUESTED** [review]
Minor changes are required

File: assets/js/local-storage-theme-check.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+document.addEventListener("DOMContentLoaded", checkTheme);
+
+function checkTheme() {
```
[the remark this answers is not in the document]
Comment (sananand007) [reply assets/js/local-storage-theme-check.js:3]: remove this file, not really needed anymore

File: assets/templates/user_profile.html (hunk 3 above)
Comment (sananand007) [reply assets/templates/user_profile.html:36]: yes in style.css

File: assets/templates/about.html [hunk 7]
```diff
@@ -47,8 +53,9 @@
       </div>
     </nav>
     <h1
-      class="inherit-line-height"
-      style="color: #24c1e0; text-align: center; background-color: whitesmoke"
+    class="inherit-line-height"
```
Comment (timwangmusic) [inline assets/templates/about.html:56]: Run formatter on this file please

**Review by timwangmusic — APPROVED** [review]
LGTM! Nice work

File: assets/templates/user_profile.html (hunk 4 above)
Comment (sananand007) [reply assets/templates/user_profile.html]: done

File: assets/templates/about.html (hunk 7 above)
Comment (sananand007) [reply assets/templates/about.html]: done

**Pull request merged**